alpenhorn/util.py

```diff
@@ -111,6 +98,46 @@ def get_hostname() -> str:
     return socket.gethostname().split(".")[0]
 
 
+def pretty_bytes(num: int) -> str:
+    """Return a nicely formatted string describing a size in bytes.
+
+    Parameters
+    ----------
+    num : int
+        Number of bytes
+
+    Returns
+    -------
+    pretty_bytes : str
+        A formatted string using power-of-two prefixes,
+        e.g. "103.4 GiB"
+
+    Raises
+    ------
+    OverflowError
+        `num` was greater than or equal to 2 ** 70.
+    """
+
+    # Reject weird stuff
+    try:
+        if num < 0:
+            raise ValueError("negative size")
+    except TypeError:
+        raise TypeError("non-numeric size")
+
+    num = int(num)
+    if num < 2**10:
+        return f"{num} B"
+
+    for x, p in enumerate("kMGTPE"):
+        if num < 2 ** ((2 + x) * 10):
+            num /= 2 ** ((1 + x) * 10)
+            return f"{num:.1f} {p}iB"
+
+    # overflow
+    raise OverflowError("a suffusion of yellow")
```

Comment:
I like it but the exception should _probably_ be a bit more descriptive...

Reply:
Is it even worth having an exception here?  This function is only ever used for logging.  Do we want alpenhorn crashing when it discovers it's logging nonsense?

Reply:
We probably want something to happen if it's logging garbage so that someone notices it sooner rather than later, like an alert of some sort, but I'm not really sure how we'd implement that. If it is logging garbage it might be indicative of a larger issue, so if it eventually crashes from something else it would be rough to go back and find out that we can't use the logs to figure out what happened...

Reply:
In the end I just removed this exception.  I think it's best just to go on with logging in this case and let whatever caused the unreasonably large number do the crashing, if that needs to happen.